File: btc.js
const bitcoinjs = require('bitcoinjs-lib');
const crypto = require('crypto');

function private_key_to_public_key(privKeyHex) {
  const privateKey = bitcoinjs.ECPair.fromPrivateKey(Buffer.from(privKeyHex, 'hex'));
  return privateKey.getPublicKey().encodeCompressed('hex');
}

function pubkey_to_address(pubKey, magicByte = 0) {
  const pubKeyBuffer = Buffer.from(pubKey, 'hex');
  const sha256 = crypto.createHash('sha256');
  const ripemd160 = crypto.createHash('ripemd160');
  sha256.update(pubKeyBuffer);
  ripemd160.update(sha256.digest());
  const address = bitcoinjs.base58check.encode(Buffer.concat([Buffer.from([magicByte]), ripemd160.digest()]));
  return address;
}

// Replace '' as private key
const privateKeyWIF = '';
const privateKeyHex = bitcoinjs.base58check.decode(privateKeyWIF).toString('hex').slice(2); // Decode WIF and remove '80' prefix
console.log("Private key (hex):", privateKeyHex);

const publicKey = private_key_to_public_key(privateKeyHex);
console.log("Public key (hex compressed):", publicKey);

const address = pubkey_to_address(publicKey);
console.log("Compressed Bitcoin address (base58check):", address);

File: btc.py
import bitcoin, hashlib, binascii, base58

def private_key_to_public_key(privKeyHex: str) -> (int, int):
    privateKey = int(privKeyHex, 16)
    return bitcoin.fast_multiply(bitcoin.G, privateKey)

def pubkey_to_address(pubKey: str, magic_byte = 0) -> str:
    pubKeyBytes = binascii.unhexlify(pubKey)
    sha256val = hashlib.sha256(pubKeyBytes).digest()
    ripemd160val = hashlib.new('ripemd160', sha256val).digest()
    return bitcoin.bin_to_b58check(ripemd160val, magic_byte)

# Replace '' as private key
private_key_wif = ''
private_key_hex = base58.b58decode_check(private_key_wif).hex()[2:]  # Decode WIF and remove '80' prefix
print("Private key (hex):", private_key_hex)

public_key = private_key_to_public_key(private_key_hex)
print("Public key (x,y) coordinates:", public_key)

compressed_public_key = bitcoin.compress(public_key)
print("Public key (hex compressed):", compressed_public_key)

address = pubkey_to_address(compressed_public_key)
print("Compressed Bitcoin address (base58check):", address)